libraries/aws/shadow/src/aws_iot_shadow_api.c

```diff
@@ -1133,77 +1136,90 @@ AwsIotShadowError_t AwsIotShadow_Wait( AwsIotShadowOperation_t operation,
                                        const char ** const pShadowDocument,
                                        size_t * const pShadowDocumentLength )
 {
-    IOT_FUNCTION_ENTRY( AwsIotShadowError_t, AWS_IOT_SHADOW_STATUS_PENDING );
+    /*AwsIotShadowError_t status = AWS_IOT_SHADOW_STATUS_PENDING; */
+    AwsIotShadowError_t status = AWS_IOT_SHADOW_SUCCESS;
 
     /* Check that AwsIotShadow_Init was called. */
     if( _checkInit() == false )
     {
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_SHADOW_NOT_INITIALIZED );
+        status = AWS_IOT_SHADOW_NOT_INITIALIZED;
     }
 
-    /* Check that reference is set. */
-    if( operation == NULL )
+    if( status == AWS_IOT_SHADOW_SUCCESS )
```

Comment:
The blocks here could be structured as `if`-`else if`-`else-if`-`else`

Reply:
Good point.